### PR TITLE
Feature/components/product list

### PR DESCRIPTION
**Pull request by Vomcat**

### CODE REVIEW

File: src/pages/Homepage.tsx [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import Banner from "../components/Banner/Banner";
```
Comment (magnowak) [inline src/pages/Homepage.tsx:1]: absolute import

File: src/components/layout/tests/ButtonPlus.test.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import { render } from "@testing-library/react";
+import renderer from "react-test-renderer";
```
Comment (magnowak) [inline src/components/layout/tests/ButtonPlus.test.tsx:2]: this renderer is necessary to test styled components' css, yes?

File: src/components/Product/tests/ProductList.test.tsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  {
+    id: 2,
+    image: "/assets/images/item2.png",
+    name: "Mniam Mniam Food as adsdasdasd",
+    price: 5,
+  },
+];
+
+describe("Product List render test", () => {
+  it("renders correctly", async () => {
```
Comment (magnowak) [inline src/components/Product/tests/ProductList.test.tsx:21]: 🥇

File: src/components/Product/tests/ProductList.test.tsx [hunk 4]
```diff
@@ -0,0 +1,35 @@
+import { render, screen } from "@testing-library/react";
+
+import ProductList from "../ProductList";
```
Comment (magnowak) [inline src/components/Product/tests/ProductList.test.tsx]: absolute import

File: src/components/Product/ProductList.tsx [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    }
+  };
+
+  useEffect(() => {
+    fetchData();
+  }, []);
+
+  return (
+    <ProductListWrapper>
+      {items.map((item) => (
```
Comment (magnowak) [inline src/components/Product/ProductList.tsx:47]: you could use destructuring assignment here, like so:
`
items.map((item)=>{
    const {id, name, price, image} = item;
    return (
         <Item
            key={id}
            id
            name
            price
          image
        />
    )
}):
`

Comment (magnowak) [reply src/components/Product/ProductList.tsx:47]: your way is also clear and concise so use whichever you prefer ;)

**Review by magnowak — APPROVED** [review]
Looks good

File: src/interfaces/Product.ts [hunk 6]
```diff
@@ -0,0 +1,6 @@
+export interface ProductItemElements {
+  id: number;
+  image?: string;
```
Comment (magnowak) [inline src/interfaces/Product.ts]: if it is a string then I would call the property "imageURI" (uri not url). Also have you checked if ProductItem will render correctly without an image?

File: src/interfaces/Product.ts [hunk 7]
```diff
@@ -0,0 +1,6 @@
+export interface ProductItemElements {
```
Comment (magnowak) [inline src/interfaces/Product.ts]: I would call it "ProductItemProps"

File: src/components/layout/tests/ButtonPlus.test.tsx [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+import "jest-styled-components";
+
+import ButtonPlus from "../ButtonPlus";
```
Comment (magnowak) [inline src/components/layout/tests/ButtonPlus.test.tsx]: absolute path

File: src/components/Product/ProductItem.tsx [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+const ProductItem: React.FC<ProductItemElements> = ({
+  id,
+  image,
+  name,
+  price,
+}) => {
+  return (
+    <Item>
+      <ItemImage>
+        <img src={image} alt="elo" />
```
Comment (magnowak) [inline src/components/Product/ProductItem.tsx]: you sure alt should be "elo"? :D

**Review by TomaszSack — APPROVED** [review]
looks good

File: src/pages/Homepage.tsx (hunk 1 above)
Comment (Vomcat) [reply src/pages/Homepage.tsx:1]: Fixed

File: src/components/layout/tests/ButtonPlus.test.tsx (hunk 2 above)
Comment (Vomcat) [reply src/components/layout/tests/ButtonPlus.test.tsx:2]: yes

File: src/components/Product/tests/ProductList.test.tsx (hunk 4 above)
Comment (Vomcat) [reply src/components/Product/tests/ProductList.test.tsx]: Fixed

File: src/interfaces/Product.ts (hunk 6 above)
Comment (Vomcat) [reply src/interfaces/Product.ts]: Image is required so i changed it

File: src/interfaces/Product.ts (hunk 7 above)
Comment (Vomcat) [reply src/interfaces/Product.ts]: I change it

File: src/components/layout/tests/ButtonPlus.test.tsx (hunk 8 above)
Comment (Vomcat) [reply src/components/layout/tests/ButtonPlus.test.tsx]: Fixed

File: src/components/Product/ProductItem.tsx (hunk 9 above)
Comment (Vomcat) [reply src/components/Product/ProductItem.tsx]: No xd

**Pull request merged**